unity-renderer/Assets/Rendering/Culling/CullingObjectsTracker.cs

```diff
@@ -1,144 +1,198 @@
-        /// </summary>
-        /// <param name="includeInactives">True for add inactive objects to the tracked list.</param>
-        public void ForcePopulateRenderersList(bool includeInactives)
+        private void ForceCleanRendereables()
         {
-            List<Renderer> renderersList = new List<Renderer>();
-            List<SkinnedMeshRenderer> skinnedRenderersList = new List<SkinnedMeshRenderer>();
-            Renderer[] allRenderers = Object.FindObjectsOfType<Renderer>(includeInactives);
+            for (int i = rendereables.Count - 1; i >= 0; i--)
+            {
+                if (rendereables[i] == null || rendereables[i].container == null)
+                    rendereables.RemoveAt(i);
+            }
+        }
 
-            foreach (Renderer renderer in allRenderers)
+        private IEnumerator CalculateRenderers()
+        {
+            int amount = 0;
+            List<Rendereable> checkingRendereables = new List<Rendereable>(rendereables);
+            usingRenderers.Clear();
+            foreach (Rendereable rendereable in checkingRendereables)
             {
-                if ((((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
-                    && !(renderer is ParticleSystemRenderer))
+                if (rendereable == null || !rendereable.container)
+                {
+                    rendereables.Remove(rendereable);
+                    continue;
+                }
+
+                if (!rendereable.container.activeInHierarchy)
+                    continue;
+
+                foreach (Renderer renderer in rendereable.renderers)
                 {
-                    if (renderer is SkinnedMeshRenderer)
-                        skinnedRenderersList.Add((SkinnedMeshRenderer)renderer);
-                    else
-                        renderersList.Add(renderer);
+                    if (renderer == null || !renderer.gameObject.activeInHierarchy)
+                    {
+                        amount++;
+                        continue;
+                    }
+
+                    if (amount >= CullingControllerSettings.MAX_POPULATING_ELEMENTS_PER_FRAME)
+                    {
+                        yield return null;
+                        amount = 0;
+                    }
+                    amount++;
+
+                    if (((1 << renderer.gameObject.layer) & ignoredLayersMask) == 0)
+                        usingRenderers.Add(renderer);
                 }
             }
-            renderers = renderersList;
-            skinnedRenderers = skinnedRenderersList;
         }
 
-        public void SetIgnoredLayersMask(int ignoredLayersMask) { this.ignoredLayersMask = ignoredLayersMask; }
+        private IEnumerator CalculateSkinnedRenderers()
+        {
+            int amount = 0;
+            List<SkinnedMeshRenderer> checkingSkinnedRenderers = new List<SkinnedMeshRenderer>(detectedSkinnedRenderers);
+            usingSkinnedRenderers = new List<SkinnedMeshRenderer>();
```

Comment:
What is the purpose of creating a new instance? I believe you can just `Clear` right the same collection over and over again.

Reply:
Tests where failing if I did a `Clear`, so I decided for this approach

Reply:
what was the reason of failiure?

Reply:
Tests did not recognize the assignation of the variable declaration on top of the class, I think in a later task where we add new tests this should be revised